src/stream.js

```diff
@@ -97,16 +134,18 @@ module.exports = ({ id, name, send, onEnd = () => {}, type = 'initiator', maxMsg
           }
         }
       } catch (err) {
-        // Send no more data if this stream was remotely reset
-        if (err.code === ERR_MPLEX_STREAM_RESET) {
-          log('%s stream %s reset', type, name)
-        } else {
-          log('%s stream %s error', type, name, err)
-          send({ id, type: Types.RESET })
-        }
+        if (err.code !== MPLEX_WRITE_STREAM_CLOSED) {
+          // Send no more data if this stream was remotely reset
+          if (err.code === ERR_MPLEX_STREAM_RESET) {
+            log('%s stream %s reset', type, name)
+          } else {
+            log('%s stream %s error', type, name, err)
+            send({ id, type: Types.RESET })
```

Comment:
This might result in sending attempts when the sink ended in the other party (read closed). This will result in `ERR_STREAM_DOESNT_EXIST` error thrown.
Better be safe. Can you create `_send` function that checks if `sinkEnded` is false before doing the actual send? This will guarantee that on going writes will be stopped if we receive a message that the other party closed the read stream

Reply:
Interesting, I have a couple of meetings now. I can recheck later. Otherwise, we can revisit if we should throw the error. I need to properly test this with libp2p first